File: fixtures/scale/lua/large/checks/check_ringshift.lua
local ringshift = require("ciphers.ringshift")
local ringshift_spec = require("specs.ringshift_spec")
local corpus_rotate = require("corpus.corpus_rotate")

local function check_ringshift()
  local spec = ringshift_spec.get()
  for _, text in ipairs(corpus_rotate.corpus_rotate()) do
    local sealed = ringshift.ringshift_encrypt(text, spec.key)
    local opened = ringshift.ringshift_decrypt(sealed, spec.key)
    if opened ~= text then return false end
  end
  return spec.category == "rotate"
end

return { check_ringshift = check_ringshift }

File: fixtures/scale/typescript/large/src/checks/CheckRingshift.ts
import { Ringshift } from '../ciphers/Ringshift';
import { RingshiftSpec } from '../specs/RingshiftSpec';
import { CorpusRotate } from '../corpus/CorpusRotate';

export class CheckRingshift {
  static run(): boolean {
    const spec = RingshiftSpec.get();
    for (const text of CorpusRotate.texts()) {
      const sealed = Ringshift.encrypt(text, spec.key);
      const opened = Ringshift.decrypt(sealed, spec.key);
      if (opened !== text) return false;
    }
    return spec.category === 'rotate';
  }
}
